src/core/util/options.js

```diff
@@ -248,6 +249,30 @@ function normalizeProps (options: Object) {
 }
 
 /**
+ * Normalize all injections into Object-based format
+ */
+function normalizeInject (options: Object) {
+  const inject = options.inject
+  if (!inject) return
```

Comment:
In addition, with normalization here, we should be able to simplify the `resolveInject` function here since the normalized value will always be objects: https://github.com/vuejs/vue/blob/dev/src/core/instance/inject.js#L38